src/server.rs

```diff
@@ -64,7 +64,9 @@ where
 /// Given a host and port, creates a string of the form "host:port" or
 /// "[host]:port", depending on whether the host is an IPv6 literal.
 fn join_host_port(host: &str, port: u16) -> String {
-    if let Ok(ip) = host.parse::<IpAddr>() {
+    if host.starts_with("unix:") {
```

Comment:
The return value must be nul-terminated, so how about use `CString` instead?

Reply:
There may be two allocations, one is for format, second is for cstring.